### PR TITLE
feat(super-flags): Add GetPath method in superflags

### PR DESCRIPTION
**Pull request by ahsanbarkati**
In super-flags, the file-paths are given within quotes. So the shell is not able to expand the paths having `~` to `/home/user/...`.
This PR adds `GetPath()` method on super flags which expands the paths having `~` to `/home/user/..`.

For example:
`dgraph alpha --acl="secret_file=~/hmac-secret"` 

the above command throws an error that path doesn't exists even if it exists.

---
This change is [Reviewable](https://reviewable.io/reviews/dgraph-io/ristretto/258)

### CODE REVIEW

**Review by karlmcguire — APPROVED** [review]
:lgtm:

> Reviewed 2 of 2 files at r1.
> *[Reviewable](https://reviewable.io/reviews/dgraph-io/ristretto/258#-:-MVMXO2t5DdmU-UlZMMt:bnfp4nl)* status: :shipit: complete! all files reviewed, all discussions resolved (waiting on @jarifibrahim, @manishrjain, and @martinmr)

**Comment by karlmcguire** [comment]
Does this address row #34 in Haqathon sheet?

**Comment by ahsanbarkati** [comment]
> Does this address row #34 in Haqathon sheet? If not, might be a good place to improve that.

If we want to support filenames with semi-colons then we will have to change the separator itself. Not sure if we expect filenames with semi-colons.

File: z/flags.go [hunk 1]
```diff
@@ -247,3 +249,34 @@ func (sf *SuperFlag) GetString(opt string) string {
...
+func (sf *SuperFlag) GetPath(opt string) string {
+	p := sf.GetString(opt)
+	path, err := expandPath(p)
+	if err != nil {
+		log.Fatalf("Failed to get path: %+v", err)
+	}
+	return path
+}
+
+// expandPath expands the paths containing ~ to /home/user. It also computes the ablosule path
```
Comment (jarifibrahim) [inline z/flags.go:262]: typo ablosule => absolute

**Review by jarifibrahim — APPROVED** [review]

File: z/flags.go [hunk 2]
```diff
@@ -247,3 +249,34 @@ func (sf *SuperFlag) GetString(opt string) string {
...
+// from the relative paths. For example: ~/abc/../cef will be transformed to /home/user/cef.
+func expandPath(path string) (string, error) {
+	if len(path) == 0 {
+		return "", nil
+	}
+	if path[0] == '~' {
```
Comment (ajeetdsouza) [inline z/flags.go:268]: This does not take into account filenames that start with `~`, eg. `"~myfilename"`.

On UNIX,
```go
path[0] == '~' && (len(path) == 1 || path[1] == '/')
```

On Windows,
```go
path[0] == '~' && (len(path) == 1 || path[1] == '/' || path[1] == '\\')
```

**Review by ajeetdsouza — CHANGES_REQUESTED** [review]

File: z/flags.go (hunk 1 above)
Comment (ahsanbarkati) [reply z/flags.go]: Done

File: z/flags.go (hunk 2 above)
Comment (ahsanbarkati) [reply z/flags.go]: We can actually create file with names `'~\\filename'` in unix. Which will cause problems if we include windows fix. I have fixed this for unix, but not for windows.

Comment (ajeetdsouza) [reply z/flags.go]: @ahsanbarkati you can use `os.IsPathSeparator` to do this in a cross-platform manner.

Comment (ahsanbarkati) [reply z/flags.go]: Nice. Thanks. Done :)

**Pull request merged**